src/modules/deploy/provider/BaseProvider.ts

```diff
@@ -0,0 +1,15 @@
+export type Functions = Record<string, { url: string }>
+
+export abstract class BaseProvider {
+  public account: string
+
+  constructor(account: string) {
+    this.account = account
+  }
+
+  public abstract async createOrUpdateFunction(functionName: string, content: Buffer): Promise<void>
```

Comment:
Think about the most appropriate return type.

Reply:
It would be really nice to return the URL, but we are not persisting that information right now. It's only available when we create an API Gateway, so the update method doesn't have access to that data.